graphx/src/main/scala/org/apache/spark/graphx/Pregel.scala

```diff
@@ -139,6 +141,14 @@ object Pregel extends Logging {
       // get to send messages. We must cache messages so it can be materialized on the next line,
       // allowing us to uncache the previous iteration.
       messages = g.mapReduceTriplets(sendMsg, mergeMsg, Some((newVerts, activeDirection))).cache()
+
+      if (checkpoint && i % checkpointFrequency == checkpointFrequency - 1) {
```

Comment:
We should output a log if the checkpoint is not active when the checkpoint dir is not set